src/hera/workflows/v5/_context.py

```diff
@@ -17,15 +17,11 @@ def enter(self, p: Subbable) -> None:
         self._pieces.append(p)
 
     def exit(self) -> None:
-        if len(self._pieces) == 1:
-            return  # only the workflow is in the context
+        self._pieces.pop()
 
-        popped = self._pieces.pop()
-        main = self._pieces[0]
-        main._add_sub(popped)
-
-    def add_sub_node(self, node: Union[_SubNodeMixin, TTemplate, _DAGTaskMixin]) -> None:
-        self._pieces[-1]._add_sub(node)
+    def add_sub_node(self, node: Union[_SubNodeMixin, TTemplate]) -> None:
+        if self._pieces:
+            self._pieces[-1]._add_sub(node)
```

Comment:
Related to https://github.com/argoproj-labs/hera-workflows/pull/449#discussion_r1125715224